service/controller/resource/monitoring/scrapeconfigs/resource.go

```diff
@@ -124,7 +128,7 @@ func toSecret(ctx context.Context, v interface{}, config Config) (*corev1.Secret
 	}
 	config.WorkloadClusterETCDDomain = workloadClusterETCDDomain
 
-	scrapeConfigs, err := toData(v, config)
+	scrapeConfigs, err := toData(ctx, config.K8sClient.CtrlClient(), v, config)
```

Comment:
Had to pass the ctrl client from here because it's impossible to mock otherwise

Reply:
Done